packages/cms-lib/lib/errorHandlers/customErrors.js

```diff
@@ -0,0 +1,14 @@
+const getErrorMessage = error => {
+  const { category } = error;
+
+  switch (category) {
+    case 'MISSING_SCOPES':
```

Comment:
I do wonder if we should somehow add this handling closer to the API being called so that we can provide clearer context based on the API being called.

Reply:
We could add a handler to [the catch block](https://github.com/HubSpot/hubspot-cms-tools/blob/62597483e8eda34d60578331200f72779a2e0262/packages/cms-cli/commands/secrets.js#L66) of `configureSecretsAddCommand` that would display a custom message instead of calling `logApiErrorInstance` and we would have access to the entire error context.

Reply:
Is there a different approach that could work? Perhaps somehow the API helpers could be wrapped somehow? It feels to me like we should be able to handle this in the CLI in a similar fashion to how we do it in the UI where either there are upfront scope checks for certain situations or there are specific error handlers for specific contexts. For example, there are situations where a 404 has a specific meaning that could be conveyed more clearly to the user when we know the context.